multi-educator/src/Dscp/MultiEducator/Launcher/Runner.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE OverloadedLabels #-}
+
+-- | Helpers for starting an Educator node
+
+module Dscp.MultiEducator.Launcher.Runner
+    ( formEducatorContext
+    , launchEducatorRealMode
+    ) where
+
+import Loot.Config (option, sub)
+import Loot.Log (MonadLogging)
+
+import Dscp.Config
+import Dscp.MultiEducator.Config
+import Dscp.MultiEducator.Launcher.Mode (MultiEducatorContext (..), MultiEducatorRealMode)
+import Dscp.MultiEducator.Launcher.Resource (MultiEducatorResources (..))
+import Dscp.Resource.Class (AllocResource (..), InitParams (..))
+import Dscp.Resource.Functions
+import Dscp.Rio (runRIO)
+import Dscp.Witness.Launcher (formWitnessContext)
+
+-- | Make up Educator context from dedicated pack of allocated resources.
+formEducatorContext
+    :: (MonadIO m, MonadCatch m, MonadLogging m, HasMultiEducatorConfig)
+    => MultiEducatorResources
+    -> m MultiEducatorContext
+formEducatorContext _mecResources = do
+    let wConfig = rcast multiEducatorConfig
+    _mecWitnessCtx <- withWitnessConfig wConfig $
+        formWitnessContext (_merWitnessResources _mecResources)
+    pure MultiEducatorContext{..}
```

Comment:
... or just `bfnSomething` names, because you don't start your accessor names with `_`.

Both facts will confuse code readers. I'm advocating for using lenses and prisms instead of matching with `Foo { bar }`, or, at least, matching with `Foo { bar = theBar }`.

Reply:
I'm against usage of `{..}`-extension here. It makes reading the code a lot worse.

Reply:
Well, this code is just as well from the original educator code.